xpclr/util.py

```diff
@@ -107,39 +104,47 @@ def get_sample_ids(sample_input):
     return samples
 
 
-def load_vcf_wrapper(path, seqid, samples, samples_path):
+def load_vcf_wrapper(path, seqid, samples, samples_path, gdistkey):
 
+    var_gdistkey = 'variants/{0}'.format(gdistkey)
     callset = allel.read_vcf(
         path,
         region=seqid,
-        fields=['variants/POS', 'calldata/GT', 'samples'],
+        fields=['variants/POS', var_gdistkey, 'calldata/GT', 'samples'],
+        numbers = {var_gdistkey : 1},
         tabix="tabix",
         samples=samples)
 
     assert "samples" in callset.keys(), "None of the samples provided in {0!r} are found in {1!r}".format(
         samples_path, path)
-
+    
     p = allel.SortedIndex(callset["variants/POS"])
     g = allel.GenotypeArray(callset['calldata/GT'])
```

Comment:
TODO: find better approach that doesn't read everything into memory here.

Reply:
What about chunked or dask arrays? I haven't worked with them much, but I think they have the advantage of not loading everything into memory.

Reply:
Yes- I've used them quite a bit. I think that's the easiest solution, though it will require passing the `.compute` method downstream where appropriate. Beyond scope for this PR though.